.github/workflows/ci.yml

```diff
@@ -190,83 +157,52 @@ jobs:
-        uses: actions/checkout@v2
-        with:
-          repository: pwang7/fuse-xfstests
-          path: ./fuse-xfstests
-      #- name: Run fuse-xfstests
-      #  run: |
-      #    sudo apt update
-      #    sudo apt install xfslibs-dev uuid-dev libtool-bin \
-      #        e2fsprogs automake gcc libuuid1 quota attr libattr1-dev make \
-      #        libacl1-dev libaio-dev xfsprogs libgdbm-dev gawk fio dbench \
-      #        uuid-runtime python sqlite3 # liburing-dev
-      #    sudo useradd -m fsgqa
-      #    sudo groupadd fsgqa || true
-      #    sudo useradd 123456-fsgqa
-      #    sudo useradd fsgqa2
-      #    cd fuse-xfstests
-      #    make
-      #    #sudo ./check-datenlord
-      #    cd -
+      - name: Enable bind_mounter permission
+        run: |
+          sudo chown root:root $BIND_MOUNTER
+          sudo chmod u+s $BIND_MOUNTER
+          ls -lsh $BIND_MOUNTER
+          export ETCD_END_POINT=127.0.0.1:2379
+          export BIND_MOUNTER=`realpath $BIND_MOUNTER`
       - name: Run etcd service
         run: |
-          docker pull $ETCD_IMAGE
-          docker run -d --rm --net host --name $ETCD_CONTAINER_NAME $ETCD_IMAGE
-          docker ps
-          docker logs $ETCD_CONTAINER_NAME
-      - name: Run ntfs-3g-pjd-fstest
+          ./scripts/setup_etcd.sh
+      - name: Modify fuse.conf
+        run: sudo sed -i 's/#user_allow_other/user_allow_other/g' /etc/fuse.conf
+      - name: Run Datenlord in the background locally.
+        run: |
+          ./scripts/start_local_node.sh &
+      - name: Build fstest
         run: |
-          git clone https://git.code.sf.net/p/ntfs-3g/pjd-fstest ntfs-3g-pjd-fstest
-          cd ntfs-3g-pjd-fstest
-          cp tests/conf tests/conf.bak
-          cat >tests/conf <<END
-            # fstest configuration file
-            # Known operating systems: FreeBSD, SunOS, Linux
-            os=`uname`
-            # Known file systems: UFS, ZFS, ext3, ext4, ntfs-3g, xfs, btrfs, glusterfs
-            fs="ext4"
-          END
-          cat tests/conf
-          gcc -Wall -DHAS_TRUNCATE64 -DHAS_STAT64 fstest.c -o fstest
-          cd -
-          mkdir fuse_mount_dir
-          sudo RUST_LOG=debug target/debug/datenlord start_async_fuse --mountpoint=./fuse_mount_dir --volume_info="fuse-test-bucket;http://127.0.0.1:9000;test;test1234" --volume_type=none --etcd=127.0.0.1:2379 --serverport=8800 --nodeid=test --nodeip=127.0.0.1 --capacity=1073741824 &>datenlord.log &
-          cd fuse_mount_dir
-          sudo prove -rv ../ntfs-3g-pjd-fstest/tests/chflags
-          sudo prove -rv ../ntfs-3g-pjd-fstest/tests/xacl
-      - name: Setup tmate session
-        if: ${{ failure() }}
-        uses: mxschmitt/action-tmate@v3
+          cd /home/runner/work/datenlord/datenlord/tests/fstest_dir/
+          make
+      - name: Run fstest
+        run: |
+          cd $DATENLORD_LOCAL_BIND_DIR
+          sudo prove -rv /home/runner/work/datenlord/datenlord/tests/fstest_dir/tests/
```

Comment:
Why does fstest require the root permission?

Reply:
It will use root permissions to simulate access by different users.
```
expect 0 -u 65534 -g 65534 symlink test ${n0}/${n1}
```